src/CMakeLists.txt

```diff
@@ -314,11 +314,19 @@ else()
 	endif()
 endif()
 
-# On clang and gcc, some functionalities of std::atomic require -latomic.
+# On non-recent clang and gcc, some functionalities of std::atomic require -latomic.
 # See <https://en.cppreference.com/w/cpp/atomic/atomic#Notes>.
-if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang"
+# Use -latomic if it works
+# Note that find_library does not reliably find it so we have to resort to this.
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang"
 		OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
the whole idea is to get rid of the hard-coded compiler check

Reply:
I know, but all attempts failed so far, either with linux clang or windows

is
`if (NOT WINDOWS)`
better?

Reply:
I'll commit a change and let's see if it works.

Reply:
Seems to work now. Can you test on your macOS machine?